### PR TITLE
Servo mixer rework to allow real output scaling

### PR DESCRIPTION
**Pull request by DzikuVx**

### CODE REVIEW

File: src/main/flight/servos.c [hunk 1]
```diff
@@ -234,6 +237,29 @@ void servosInit(void)
...
+     */ 
+    for (uint8_t i = 0; i < servoRuleCount; i++) {
+        servoMetadata[currentServoMixer[i].targetChannel].rateSum += abs(currentServoMixer[i].rate);
+    }
+
+    /*
+     * Compute scaling factor for upper and lower servo throw
+     */ 
+    for (uint8_t i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
+        const float rateUs = servoMetadata[i].rateSum / 100.0f * 500;
```
Comment (digitalentity) [inline src/main/flight/servos.c]: This is basically dynamic mixer wights to make sure no combination of mixer inputs saturates the servo, right?

**Review by digitalentity — COMMENTED** [review]
Overall looks good

File: src/main/flight/servos.c (hunk 1 above)
Comment (DzikuVx) [reply src/main/flight/servos.c]: yes, this is the case. So we know max theoretical value that can be written to servo and scale from this value to max/min value we set for servo

File: src/main/flight/servos.c [hunk 2]
```diff
@@ -428,15 +423,14 @@ void servoMixer(void)
...
-            }
-        }
+        /*
+         * Apply mixer speed limit. 1 [one] speed unit is defined as 10us/s: 
+         * 0 = no limiting
+         * 1 = 10us/s -> full servo sweep (from 1000 to 2000) is performed in 100s 
+         * 10 = 100us/s -> full sweep (from 1000 to 2000)  is performed in 10s
+         * 100 = 1000us/s -> full sweep in 1s
+         */
+        currentOutput[i] = (int16_t) rateLimitFilterApply4(&servoSpeedLimitFilter[i], input[from], currentServoMixer[i].speed * 10, dT);
```
Comment (digitalentity) [inline src/main/flight/servos.c]: So much nicer than the old code! :+1:

**Comment by DzikuVx** [comment]
bloody NAZE...

**Comment by DzikuVx** [comment]
@digitalentity tested statically on a bench. Looks good, including servo speed that finally works like expected. Take a look and if nothing very wrong is found, I'd say it's ready to merge. 

Fixed wingers and glider guys will be happy with finally working differential ailerons

**Comment by stronnag** [comment]
I tried this on a tethered tricopter.

With the old syntax `smix reverse 5 2 r`, the tail servo works as expected. With the machine tethered, increasing throttle to > hover results in a 'still' tail servo. No uncommanded yaw movement.

With the new equivalent (I guess) `smix 0 5 2 -100 0` as throttle is increased the tail servo oscillates from full deflection in one direction, to full deflection in the other direction (but quite slowly, around 0.5 - 1Hz).

**Comment by DzikuVx** [comment]
OK, will look into it too in that case. Would be nice have possibility to revers smix rule with rate/weight

**Comment by DzikuVx** [comment]
@stronnag please share not working diff since I was able to make servo reversed with negative smix rate and it worked just fine

**Comment by DzikuVx** [comment]
@stronnag this is irrelevant... it turned out that `smix` are loaded only when custom mixer is used. In any other case predefined smix is used that is always rate `100`.
Smix reverse with negative rate is impossible on non-custom mixer.

**Comment by DzikuVx** [comment]
OK, flight tested, works good. PIFF tuning needs update, from my tests Pgain, Igain and FF gains of flying wings are now too low. Traditional airplanes should be fine, but in both cases retuning is hightly suggested.

I had another idea for improvement, but then well, it might be just too complicated and I think it is ready to merge after all

**Pull request merged**